src/player_comptask.c

```diff
@@ -1738,9 +1764,12 @@ short tool_dig_to_pos2_f(struct Computer2 * comp, struct ComputerDig * cdig, TbB
             return counter1;
         }
         // Being here means we didn't reached the destination - we must do some kind of action
-        if (slab_is_liquid(subtile_slab(gldstl_x), subtile_slab(gldstl_y)))
+        struct SlabMap* action_slb = get_slabmap_block(subtile_slab(gldstl_x), subtile_slab(gldstl_y));
+        if (slab_kind_is_liquid(action_slb->kind))
```

Comment:
This line is no longer needed. It returns 'is water or lava'. In the next line you check lava and water separately.